lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -72,6 +72,25 @@ static bool isDuplicatableNullaryExpression(Operation *op) {
   return false;
 }
 
+/// Return the verilog name of the operations that can define a symbol.
+static StringRef getSymOpName(Operation *symOp) {
+  // Typeswitch of operation types which can define a symbol.
+  return TypeSwitch<Operation *, StringRef>(symOp)
+      .Case<HWModuleOp>([&](HWModuleOp op) { return op.getName(); })
+      .Case<HWModuleExternOp>(
+          [&](HWModuleExternOp op) { return op.getVerilogModuleName(); })
+      .Case<HWGeneratorSchemaOp>(
+          [&](HWGeneratorSchemaOp op) { return op.sym_name(); })
+      .Case<InstanceOp>([&](InstanceOp op) { return op.getName().getValue(); })
+      .Case<WireOp>([&](WireOp op) { return op.name(); })
+      .Case<RegOp>([&](RegOp op) { return op.name(); })
+      .Default([&](Operation *op) {
+        return op->hasAttrOfType<StringAttr>("name")
+                   ? op->getAttrOfType<StringAttr>("name").getValue()
+                   : "";
```

Comment:
Is this last case necessary?  Per the recent discussion about sv.name etc, I'd prefer to keep this narrow.  Unless there is a client that needs this, please remove it and make it an error. (`op->emitOpError()` and `return ""`).

Reply:
Added other possible operations that can have symbols (`InterfaceOp`,...) and the default case will now error out.